File: lib/android/app/src/main/java/com/reactnativenavigation/views/element/TransitionSet.java
package com.reactnativenavigation.views.element;

import java.util.ArrayList;
import java.util.List;

public class TransitionSet {
    List<SharedElementTransition> validSharedElementTransitions = new ArrayList<>();
    List<ElementTransition> validElementTransitions = new ArrayList<>();

    public boolean isEmpty() {
        return validSharedElementTransitions.isEmpty() && validElementTransitions.isEmpty();
    }

    public void add(SharedElementTransition transition) {
        validSharedElementTransitions.add(transition);
    }

    public void add(ElementTransition transition) {
        validElementTransitions.add(transition);
    }

    public int size() {
        return validElementTransitions.size() + validSharedElementTransitions.size();
    }
}

File: lib/android/app/src/main/java/com/reactnativenavigation/views/element/TransitionSet.kt
package com.reactnativenavigation.views.element

import java.util.*

class TransitionSet {
    var validSharedElementTransitions: MutableList<SharedElementTransition> = ArrayList()
    var validElementTransitions: MutableList<ElementTransition> = ArrayList()
    val isEmpty: Boolean
        get() = validSharedElementTransitions.isEmpty() && validElementTransitions.isEmpty()
    val transitions: List<Transition>
        get() = validElementTransitions + validSharedElementTransitions

    fun add(transition: SharedElementTransition) {
        validSharedElementTransitions.add(transition)
    }

    fun add(transition: ElementTransition) {
        validElementTransitions.add(transition)
    }

    fun size(): Int {
        return validElementTransitions.size + validSharedElementTransitions.size
    }
}